lib/discovery.dart

```diff
@@ -2,6 +2,8 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+// @dart=2.7
+
 @Deprecated("Use the package_config.json based API")
```

Comment:
We should do the breaking version and delete these libs.

Reply:
Seems the analyzer is still *using* the deprecated API.

Reply:
Is that because they _need_ to use it or just haven't migrated?

Reply:
They are reading `.packages` files directly, then using the legacy functionality in this package to parse the file into a map. That's not supported directly by the new API. You can read a `.packages` file into a `PackageConfig` object using
```dart
loadPackageConfig(file, preferNewest: false);
```
The plain map from name to package root isn't supported any more because it makes no sense for `package_config.json` files.